common/src/main/java/io/apicurio/registry/rest/Headers.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2020 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.registry.rest;
+
+import io.apicurio.registry.types.ArtifactState;
+
+import java.util.function.Supplier;
+import javax.ws.rs.core.Response;
+
+/**
+ * @author Ales Justin
+ */
+public interface Headers {
+    String ARTIFACT_ID = "X-Registry-ArtifactId";
+    String VERSION = "X-Registry-Version";
+
+    String DEPRECATED = "X-Registry-Deprecated";
+
+    default void checkIfDeprecated(
+        Supplier<ArtifactState> stateSupplier,
+        String artifactId,
+        Number version,
+        Response.ResponseBuilder builder
+    ) {
+        if (stateSupplier.get() == ArtifactState.DEPRECATED) {
+            builder.header(Headers.DEPRECATED, true);
+            builder.header(Headers.ARTIFACT_ID, artifactId);
```

Comment:
Why add the artifactId and version as well?  Is this so it's available for inclusion in the log message?

Reply:
Yes.
So that any client can grab that info easily from response.
You do already have that info in some cases, when you explicitly call for a id,version tuple,
but not for latest and per-globalId.